DEVELOPMENT.md

```diff
@@ -0,0 +1,69 @@
+# Developing MXNet
+
+## Build MXNet from source with CMake
+
+It's recommended that you install CMake and Ninja.
+
+Chose the options that you want to compile with in a similar fashion as above. In particular this one is a debug CPU
+build in OSx.
+
+```
+mkdir build && cd build
+cmake -DUSE_CUDA=OFF -DUSE_OPENCV=OFF -DUSE_OPENMP=OFF -DCMAKE_BUILD_TYPE=Debug -GNinja ..
+ninja
+```
+You can check the available CMake options in CMakeLists.txt file: `cat CMakeLists.txt | grep mxnet_option`. Similar
+options are needed if you want to use CLion. Under settings you can modify the CMake options so code navigation and
+build works.
+
+Then you can use the library and install with pip. It's recommended that you use a python virtualenv or similar tool for having multiple
+installed versions and managing the python interpreter.
+
+```
+virtualenv -p /usr/bin/python3.5 py3
```

Comment:
Can you update the env name to mxnet_p36? It would help keep some new docs in alignment so we can use the same examples. Also if you end up mentioning how to create a 2.7 env, then use mxnet_p27.